yakuake-init.sh

```diff
@@ -25,24 +25,55 @@ trim ()
     echo $var
 }
 
+# debugging helper function
+#
+# usage:
+# logm "There is an ERROR o.O!"
+#
+# view log:
+# $ tail -f /var/log/syslog
+#
+logm ()
+{
+    logger $(trim ${1})
+}
+
 # Helper that sends a command to yakuake
 call_yakuake_method ()
 {
     path="${1}"
     method="${2}"
     parameters=("${@:3}")
-    result=$(dbus-send --print-reply=literal --dest=org.kde.yakuake /yakuake/${path} org.kde.yakuake.${method} "${parameters[@]}")
-    result=$(trim ${result})
 
-    if [[ -n "${result}" ]]
+    if [[ -z "${parameters}" ]]
     then
-        echo "${result}"
+	    result=$(dbus-send --print-reply=literal --dest=org.kde.yakuake /yakuake/${path} org.kde.yakuake.${method})
+	else
+		case "$method" in
+		"terminalIdsForSessionId")
+			result=$(dbus-send --print-reply=literal --dest=org.kde.yakuake /yakuake/${path} org.kde.yakuake.${method} "${parameters[@]}")
+			;;
+#		"methodX" | "methodY")
+#			do_something_else()
+#			;;
+		*)
+			result=$(dbus-send --type=method_call --dest=org.kde.yakuake /yakuake/${path} org.kde.yakuake.${method} "${parameters[@]}")
+			;;
+		esac
     fi
+
+    result=$(trim ${result})
+
+#    if [[ -n "${result}" ]]
+#    then
+        echo "${result}"
```

Comment:
Surrounding comments are no longer necessary. Please remove and adjust this line's indentation.